holoviews/core/pprint.py

```diff
@@ -255,132 +256,148 @@ def options_info(cls, plot_class, ansi=False, pattern=None):
         return '\n'.join(lines)
 
 
-class PrettyPrinter(object):
+class PrettyPrinter(param.Parameterized):
     """
     The PrettyPrinter used to print all HoloView objects via the
-    pprint classmethod.
+    pprint method.
     """
 
+    show_defaults = param.Boolean(default=False, doc="""
+        Whether to show default options as part of the repr.""")
```

Comment:
Docstring should clarify what happens when show_options=False and show_defaults=True.  Does it show nothing (probably what most would expect) or show the full set of defaults without any regard for current settings (which would be useful but may be surprising)?

Reply:
>Does it show nothing (probably what most would expect)

Right, it shows nothing, will add that once we've decided on the user level API.